internal/controllers/clusterextension_controller_test.go

```diff
@@ -1400,3 +1411,137 @@ func TestSetDeprecationStatus(t *testing.T) {
+	return nil, nil
+}
+
+// This is the function we are really testing
+func (mag *MockActionGetter) History(name string, opts ...helmclient.HistoryOption) ([]*release.Release, error) {
+	return mag.rels, mag.err
+}
+
+func (mag *MockActionGetter) Install(name, namespace string, chrt *chart.Chart, vals map[string]interface{}, opts ...helmclient.InstallOption) (*release.Release, error) {
+	return nil, nil
+}
+
+func (mag *MockActionGetter) Upgrade(name, namespace string, chrt *chart.Chart, vals map[string]interface{}, opts ...helmclient.UpgradeOption) (*release.Release, error) {
+	return nil, nil
+}
+
+func (mag *MockActionGetter) Uninstall(name string, opts ...helmclient.UninstallOption) (*release.UninstallReleaseResponse, error) {
+	return nil, nil
+}
+
+func (mag *MockActionGetter) Reconcile(rel *release.Release) error {
+	return nil
+}
+
+func TestGetInstalledBundleHistory(t *testing.T) {
+	getter := controllers.DefaultInstalledBundleGetter{}
+
+	ext := ocv1alpha1.ClusterExtension{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-ext",
+		},
+	}
+
+	mag := []MockActionGetter{
+		{
+			"No return",
+			nil, nil,
+			nil, false,
+		},
+		{
+			"ErrReleaseNotFound (special case)",
+			nil, driver.ErrReleaseNotFound,
+			nil, false,
+		},
+		{
+			"One item in history",
+			[]*release.Release{
+				{
+					Name: "test-ext",
+					Info: &release.Info{
+						Status: release.StatusDeployed,
+					},
+					Labels: map[string]string{
+						labels.BundleNameKey:      "test-ext",
+						labels.BundleVersionKey:   "1.0",
+						labels.BundleReferenceKey: "bundle-ref",
+					},
+				},
+			}, nil,
+			&controllers.InstalledBundle{
+				BundleMetadata: ocv1alpha1.BundleMetadata{
+					Name:    "test-ext",
+					Version: "1.0",
+				},
+				Image: "bundle-ref",
+			}, false,
+		},
+		{
+			"Two items in history",
+			[]*release.Release{
+				{
+					Name: "test-ext",
+					Info: &release.Info{
+						Status: release.StatusFailed,
+					},
+					Labels: map[string]string{
+						labels.BundleNameKey:      "test-ext",
+						labels.BundleVersionKey:   "2.0",
+						labels.BundleReferenceKey: "bundle-ref-2",
+					},
+				},
+				{
+					Name: "test-ext",
+					Info: &release.Info{
+						Status: release.StatusDeployed,
+					},
+					Labels: map[string]string{
+						labels.BundleNameKey:      "test-ext",
+						labels.BundleVersionKey:   "1.0",
+						labels.BundleReferenceKey: "bundle-ref-1",
+					},
+				},
+			}, nil,
+			&controllers.InstalledBundle{
+				BundleMetadata: ocv1alpha1.BundleMetadata{
+					Name:    "test-ext",
+					Version: "1.0",
+				},
+				Image: "bundle-ref-1",
+			}, false,
+		},
+	}
+
+	for _, tst := range mag {
+		t.Log(tst.description)
+		getter.ActionClientGetter = &tst
+		md, err := getter.GetInstalledBundle(context.Background(), &ext)
+		if tst.expectedError {
+			require.Error(t, err)
+		} else {
+			require.NoError(t, err)
+		}
```

Comment:
Nit: could probably simplify this to something like:
```suggestion
		require.Equal(t, err != nil, tst.expectedError, fmt.Sprintf("error state doesn't match expected. should error? %v, err: %w", tst.expectedError, err))
```

Reply:
Perhaps, not sure that's actually simpler... it's one line of code, but it takes a bit more time to grok.

Reply:
Ok, found a better way!